pkg/scalers/external_scaler.go

```diff
@@ -273,19 +329,44 @@ var connectionPoolMutex sync.Mutex
 
 // getClientForConnectionPool returns a grpcClient and a done() Func. The done() function must be called once the client is no longer
 // in use to clean up the shared grpc.ClientConn
-func getClientForConnectionPool(metadata externalScalerMetadata) (pb.ExternalScalerClient, error) {
+func getClientForConnectionPool(metadata externalScalerMetadata, logger logr.Logger) (pb.ExternalScalerClient, error) {
 	connectionPoolMutex.Lock()
 	defer connectionPoolMutex.Unlock()
 
 	buildGRPCConnection := func(metadata externalScalerMetadata) (*grpc.ClientConn, error) {
+		// FIXME: DEPRECATED to be removed in v2.13
 		if metadata.tlsCertFile != "" {
+			logger.V(1).Info("tlsCertFile in ScaleObject metadata will be deprecated in v2.12. Please use" +
+				"tlsClientCert, tlsClientKey and caCert in TriggerAuthentication instead.")
 			creds, err := credentials.NewClientTLSFromFile(metadata.tlsCertFile, "")
 			if err != nil {
 				return nil, err
 			}
 			return grpc.Dial(metadata.scalerAddress, grpc.WithTransportCredentials(creds))
 		}
 
+		tlsConfig := &tls.Config{MinVersion: tls.VersionTLS13}
```

Comment:
There are several helpers for creating this tlsConfig in [`util`](https://github.com/kedacore/keda/blob/main/pkg/util/tls_config.go), please use one of them because those helpers apply different global configs like custom CAs or minimum TLS Version for respecting global configurations